oct/pages/models/checkout_page.py

```diff
@@ -0,0 +1,143 @@
+
+
+class CheckoutOptions(Block):
+    contains = {
+        "register_account": {
+            "locator": (
+                "XPATH",
+                '//*[@id="collapse-checkout-option"]/div/div/div[1]/div[1]/label/input',
+            ),
+            "class": RadioButton,
+        },
+        "guest_checkout": {
+            "locator": (
+                "XPATH",
+                '//*[@id="collapse-checkout-option"]/div/div/div[1]/div[2]/label/input',
+            ),
+            "class": RadioButton,
+        },
+        "continue_button": {"locator": ("ID", "button-account"), "class": Clickable},
+    }
+
+
+class ReturningCustomer(Block):
+    contains = {
+        "email": {"locator": ("ID", "input-email"), "class": InputField},
+        "password": {"locator": ("ID", "input-password"), "class": InputField},
+        "login_button": {"locator": ("ID", "button-login"), "class": Clickable},
+    }
+
+    def fill_out(self, **kwargs):
+        self.test.email.fill(kwargs.get("email"))
+        self.test.password.fill(kwargs.get("password"))
+
+
+class BillingDetails(Form):
+    contains = {
+        "first_name": {"locator": ("ID", "input-payment-firstname"), "class": InputField},
+        "last_name": {"locator": ("ID", "input-payment-lastname"), "class": InputField},
+        "email": {"locator": ("ID", "input-payment-email"), "class": InputField},
+        "telephone": {"locator": ("ID", "input-payment-telephone"), "class": InputField},
+        "company": {"locator": ("ID", "input-payment-company"), "class": InputField},
+        "address_1": {"locator": ("ID", "input-payment-address-1"), "class": InputField},
+        "address_2": {"locator": ("ID", "input-payment-address-2"), "class": InputField},
+        "city": {"locator": ("ID", "input-payment-city"), "class": InputField},
+        "post_code": {"locator": ("ID", "input-payment-postcode"), "class": InputField},
+        "country": {"locator": ("ID", "input-payment-country"), "class": Dropdown},
+        "region_state": {"locator": ("ID", "input-payment-zone"), "class": Dropdown},
+        "my_delivery_and_billing_addresses": {
+            "locator": ("XPATH", "//*[@id='collapse-payment-address']/div/div[2]/label/input"),
+            "class": CheckBox,
+        },
+        "continue_button": {"locator": ("ID", "button-guest"), "class": Clickable},
+    }
+
+    def fill_out(self, **kwargs):
+        self.first_name.fill(kwargs.get("first_name"))
+        self.last_name.fill(kwargs.get("last_name"))
+        self.email.fill(kwargs.get("email"))
+        self.telephone.fill(kwargs.get("telephone"))
+        self.company.fill(kwargs.get("company"))
+        self.address_1.fill(kwargs.get("address_1"))
+        self.address_2.fill(kwargs.get("address_2"))
+        self.city.fill(kwargs.get("city"))
+        self.post_code.fill(kwargs.get("post_code"))
+        self.country.select(kwargs.get("country"))
+        self.region_state.select(kwargs.get("region_state"))
+
```

Comment:
BillingDetails is not a conventional form, since it does not have submit option. Better use Block instead.
```suggestion
class BillingDetails(Block):
```

```suggestion
self.region_state.select(kwargs.get("region_state"))
self.continue_button.click()
```